compiler/x/amd64/codegen/AMD64SystemLinkage.cpp

```diff
@@ -265,6 +265,79 @@ TR::AMD64Win64FastCallLinkage::AMD64Win64FastCallLinkage(TR::CodeGenerator *cg)
    TR_ASSERT(p == (machine()->getNumGlobalGPRs() + machine()->_numGlobalFPRs), "assertion failure");
    }
 
+// Build arguments for microsoft x64 calling convention dispatch.
+//
+int32_t TR::AMD64Win64FastCallLinkage::buildArgs(TR::Node* callNode, TR::RegisterDependencyConditions* deps)
+   {
+   static bool DuplicateParameterOnStack = feGetEnv("TR_DuplicateParameterOnStack");
+   static bool UseOldFastCallLinkage = feGetEnv("TR_UseOldFastCallLinkage");
+   if (UseOldFastCallLinkage || callNode->getOpCode().isIndirect())
+      {
+      return TR::AMD64SystemLinkage::buildArgs(callNode, deps);
+      }
+
+   // Basic information
+   static TR::RealRegister::RegNum IntegerParams[] = { TR::RealRegister::ecx,  TR::RealRegister::edx,  TR::RealRegister::r8,   TR::RealRegister::r9   };
+   static TR::RealRegister::RegNum FloatParams[]   = { TR::RealRegister::xmm0, TR::RealRegister::xmm1, TR::RealRegister::xmm2, TR::RealRegister::xmm3 };
```

Comment:
Why not use the `AMD64Win64FastCallLinkage` linkage properties that already describe this information rather than re-implementing it here?

Reply:
Thank you for bring this out and apology for missing your questions.

By using this static array, it eases the C++ compiler to conclude information. For instance, in the following code, this strategy allows:
```
static_assert(sizeof(IntegerParams) == sizeof(FloatParams), "Incorrect parameter-passing register configuration");
static const int MaxNumberOfParamsInRegister = sizeof(IntegerParams) / sizeof(IntegerParams[0]);
...
TR::Register* paramRegisters[MaxNumberOfParamsInRegister] = { 0 };
```
While the property route would require a dynamic allocation of `paramRegisters` array.

I agree duplicating this logic isn't the right way to go, I plan to change following part:
```
   _properties._argumentRegisters[p++] = TR::RealRegister::ecx;
   _properties._argumentRegisters[p++] = TR::RealRegister::edx;
   _properties._argumentRegisters[p++] = TR::RealRegister::r8;
   _properties._argumentRegisters[p++] = TR::RealRegister::r9;
```
into
````
for (size_t i = 0; i < sizeof(IntegerParams) / sizeof(IntegerParams[0]); i++)
   {
   _properties._argumentRegisters[p++] = IntegerParams[i];
   }
````
In this way, we would get the benefits of both.
However, I don't think this is related to this pull request, so that I would create a separate pull request for this purpose.